models/user.go

```diff
@@ -31,3 +31,21 @@ func SaveUser(email string, password string) error {
 	}
 	return nil
 }
+
+func FindOneUserByCondition(condition interface{}) (User, error) {
+	user := User{}
+	result := db.GetDB().Where(condition).First(&user)
+
+	return user, result.Error
+}
+
+func FindOneUserByID(id interface{}) (User, error) {
+	user := User{}
+	result := db.GetDB().First(&user, id)
+
+	return user, result.Error
+}
+
+func CheckPassword(hashedPassword string, password string) error {
```

Comment:
How about:

```suggestion
func ValidatePassword(hashedPassword string, password string) error {
```

Reply:
Fixed in 6872221091bc5b812877a3b984c61f969e14acfb